### PR TITLE
Remove node-specific crypto bits, use Node 16's WebCrypto

### LINKED ISSUE #2760: Simplify crypto by leveraging Node 16's WebCrypto implementation
migrate all the Node.js specific crypto implementations to be based on web crypto - https://nodejs.org/api/webcrypto.html

### PR DESCRIPTION
**Pull request by t3chguy**
Fixes https://github.com/matrix-org/matrix-js-sdk/issues/2760
Requires https://github.com/vector-im/element-web/pull/23506

---
Here's what your changelog entry will look like:

## ✨ Features
 * Remove node-specific crypto bits, use Node 16's WebCrypto ([\#2762](https://github.com/matrix-org/matrix-js-sdk/pull/2762)). Fixes #2760.

### CODE REVIEW

File: src/crypto/aes.ts [hunk 1]
```diff
@@ -180,8 +90,8 @@ async function encryptBrowser(data: string, key: Uint8Array, name: string, ivStr
  * @param {Uint8Array} key the encryption key to use
```
Comment (robintown) [inline src/crypto/aes.ts:90]: The doc comment above still says 'browser'

File: src/crypto/store/memory-crypto-store.ts [hunk 2]
```diff
@@ -292,17 +292,21 @@ export class MemoryCryptoStore implements CryptoStore {
         func(this.crossSigningKeys);
     }
 
-    public getSecretStorePrivateKey(txn: unknown, func: (key: IEncryptedPayload | null) => void, type: string): void {
-        const result = this.privateKeys[type];
+    public getSecretStorePrivateKey<T = IEncryptedPayload>(
```
Comment (robintown) [inline src/crypto/store/memory-crypto-store.ts:295]: Are these type parameters meant to be `extends IEncryptedPayload`, or can they really be anything?

**Review by robintown — COMMENTED** [review]
Looks fine overall

File: src/crypto/store/memory-crypto-store.ts (hunk 2 above)
Comment (t3chguy) [reply src/crypto/store/memory-crypto-store.ts:295]: They really can be anything, its entirely opaque object storage

**Review by uhoreg — APPROVED** [review]
Looks fine from a crypto standpoint.

File: src/crypto/store/memory-crypto-store.ts (hunk 2 above)
Comment (t3chguy) [reply src/crypto/store/memory-crypto-store.ts]: I have improved this using a mapped type, so now everything we store has to be typed better

**Review by robintown — APPROVED** [review]

**Pull request merged**